internal/sanitize/sanitize_test.go

```diff
@@ -0,0 +1,405 @@
+
+	// scan the oci-archive
+	sus, err = scanArchive(goodOCIArchive)
+	require.NoError(t, err, "scanning good OCI archive")
+	assert.False(t, sus, "check on good OCI archive")
+
+	// make sure the original versions can all be read without error
+	requireImageReadable(t, transports.ImageName(goodLayoutRef))
+	requireImageReadable(t, transports.ImageName(goodDirRef))
+	requireImageReadable(t, transports.ImageName(goodOCIArchiveRef))
+	requireImageReadable(t, transports.ImageName(goodDockerArchiveRef))
+
+	// sanitize them all
+	goodLayoutRel, err := filepath.Rel(contextDir, goodLayout)
+	require.NoErrorf(t, err, "converting absolute path %q to a relative one", goodLayout)
+	newGoodLayout, err := ImageName("oci", goodLayoutRel, contextDir, t.TempDir())
+	require.NoError(t, err, "sanitizing good OCI layout")
+	goodDirRel, err := filepath.Rel(contextDir, goodDir)
+	require.NoErrorf(t, err, "converting absolute path %q to a relative one", goodDir)
+	newGoodDir, err := ImageName("dir", goodDirRel, contextDir, t.TempDir())
+	require.NoError(t, err, "sanitizing good directory")
+	goodOCIArchiveRel, err := filepath.Rel(contextDir, goodOCIArchive)
+	require.NoErrorf(t, err, "converting absolute path %q to a relative one", goodOCIArchive)
+	newGoodOCIArchive, err := ImageName("oci-archive", goodOCIArchiveRel, contextDir, t.TempDir())
+	require.NoError(t, err, "sanitizing good OCI archive")
+	goodDockerArchiveRel, err := filepath.Rel(contextDir, goodDockerArchive)
+	require.NoErrorf(t, err, "converting absolute path %q to a relative one", goodDockerArchive)
+	newGoodDockerArchive, err := ImageName("docker-archive", goodDockerArchiveRel, contextDir, t.TempDir())
+	require.NoError(t, err, "sanitizing good docker archive")
+
+	// make sure the sanitized versions can all be read without error
+	requireImageReadable(t, newGoodLayout)
+	requireImageReadable(t, newGoodDir)
+	requireImageReadable(t, newGoodOCIArchive)
+	requireImageReadable(t, newGoodDockerArchive)
+
+	// mutate them all
+	badLayout := mutateDirectory(t, contextDir, goodLayout, filepath.Join(v1.ImageBlobsDir, blobDigest.Algorithm().String(), blobDigest.Encoded()))
+	badLayoutRel, err := filepath.Rel(contextDir, badLayout)
+	require.NoErrorf(t, err, "converting absolute path %q to a relative one", badLayout)
+	_, err = ImageName("oci", badLayoutRel, contextDir, t.TempDir())
+	require.ErrorIs(t, err, os.ErrNotExist, "sanitizing bad OCI layout")
+
+	badDir := mutateDirectory(t, contextDir, goodDir, blobDigest.Encoded())
+	badDirRel, err := filepath.Rel(contextDir, badDir)
+	require.NoErrorf(t, err, "converting absolute path %q to a relative one", badDir)
+	_, err = ImageName("dir", badDirRel, contextDir, t.TempDir())
+	require.ErrorIs(t, err, os.ErrNotExist, "sanitizing bad directory")
+
+	badOCIArchive := mutateArchive(t, contextDir, goodOCIArchive, filepath.Join(v1.ImageBlobsDir, blobDigest.Algorithm().String(), blobDigest.Encoded()))
+
```

Comment:
(Absolutely non-blocking: Drop this empty line for consistency? Alternatively, it might be possible to make parts of this test table-driven.)

Reply:
Dropping it.